#10-Link pages

Miladepour opened the pull request:
## Description

Link form pages

## Related to

#10 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

haroon-ali-dev reviewed (approved):
I have tested it locally. The flow works. The drop downs on the user preferences page appear in the address bar as query string parameters on the result page. The results on the results page display properly and the drop downs work properly changing what is in the address bar.

natyfromwonderland reviewed (approved):
Hi! What is the reason for removing 'Dehli' from location, please? 
     "North",
      "South",
      "Central",
      "West",
      "East",
Is it just for ease? I mean it will be replaced by the data from the database anyway?
Otherwise, the routing works.

EneVladMihai commented on client/src/pages/result/Result.js:
```diff
@@ -3,9 +3,12 @@ import rawData from "../../data/kujiHelplineSheet.json";
-  const [service,setService]=useState("Legal Aid");
-  const [location,setLocation]=useState("North");
+  let [searchParams, setSearchParams] = useSearchParams();
+  const [service,setService]=useState(searchParams.get("service"));
```
This is a good opportunity to explore implementing a custom hook.
See the official docs on [custom hooks](https://reactjs.org/docs/hooks-custom.html).

The pattern I see here is that we want to keep the state and seach params synchronised, so why don't we write our own special version of `useState` called `useSearchParamState` which combines `useState` and `useSearchParams` ?

Miladepour replied:
That is a good idea but need more for me to learn more about it. Can we do this in the next iteration?

Miladepour commented:
> Hi! What is the reason for removing 'Dehli' from location, please? "North", "South", "Central", "West", "East", Is it just for ease? I mean it will be replaced by the data from the database anyway? Otherwise, the routing works.

We removed 'Dehli' from the options to avoid any conflicts with other data, and we added 'Delhi' to all the options in the select input.
<option>{data} Delhi</option>
